src/main/java/control/FileMenuController.java

```diff
@@ -219,39 +285,114 @@ private void addTimeline(MouseEvent event) {
      * @param event the mouse event related to this event
      */
     private void timelineAdded(MouseEvent event) {
-        timelineModal.hideModal();
+        log.error(timelineModal.getCameraList().getSelectionModel().getSelectedIndex());
+        if (validateTimelineData()) {
+            timelineModal.hideModal();
+            String name = timelineModal.getNameField().getText();
+            String description = timelineModal.getDescriptionField().getText();
+            int selectedIndex = timelineModal.getCameraList().getSelectionModel()
+                                                             .getSelectedIndex();
+            Camera camera = timelineModal.getCameras().get(selectedIndex);
+            CameraTimeline timeline = new CameraTimeline(name, camera, description, null);
+            newProjectModal.getTimelines().add(timeline);
+            HBox box = new HBox();
+            box.getChildren().addAll(new Label(name), new Label(" - "), new Label(description));
+            newProjectModal.getTimelineList().getItems().add(box);
+        }
+    }
+    
+    /**
+     * Validate the data entered by the user in the modal to add a timeline.
+     * @return true is the data is legit, false otherwise
+     */
+    private boolean validateTimelineData() {
+        String errorString = "";
+        String name = timelineModal.getNameField().getText();
         String description = timelineModal.getDescriptionField().getText();
         int selectedIndex = timelineModal.getCameraList().getSelectionModel().getSelectedIndex();
-        Camera camera = timelineModal.getCameras().get(selectedIndex);
-        CameraTimeline timeline = new CameraTimeline(camera, description, null);
-        newProjectModal.getTimelines().add(timeline);
-        newProjectModal.getTimelineList().getItems().add(new Label(description));
+        
+        if (name.isEmpty()) {
+            errorString += "Please enter a timeline name\n";
+        }
+        
+        if (description.isEmpty()) {
+            errorString += "Please enter a timeline description\n";
+        }
+        
+        if (selectedIndex == -1) {
+            errorString += "Please select a camera\n";
+        }
+        
+        timelineModal.getErrorLabel().setText(errorString);
+        timelineModal.getErrorLabel().setTextFill(Color.RED);
+        
+        return errorString.isEmpty();
     }
     
     /**
      * Event handler for when the create project button is clicked in the NewProjectModalView.
      * @param event the mouse event related to this event
      */
     private void createProject(MouseEvent event) {
```

Comment:
This method is quite long but has no whitespaces and/or comments, could you add a few for readability